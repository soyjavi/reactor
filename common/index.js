import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

export { LAYOUT, SHAPE, THEME };
