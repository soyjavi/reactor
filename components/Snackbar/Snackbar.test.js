import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import { Snackbar } from '..';

const LOREM =
  'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s.';
const DEFAULTS = { visible: true };
const { COLOR } = THEME;

jest.useFakeTimers();

describe('<Snackbar>', () => {
  it('renders', () => {
    const tree = renderer.create(<Snackbar {...DEFAULTS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {caption}', () => {
    const tree = renderer.create(<Snackbar {...DEFAULTS} caption={LOREM} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {button}', () => {
    const tree = renderer.create(<Snackbar {...DEFAULTS} button="CLOSE" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {children}', () => {
    const tree = renderer.create(<Snackbar {...DEFAULTS}>{LOREM}</Snackbar>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Snackbar {...DEFAULTS} color={COLOR.ALERT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {visible} (false)', () => {
    const tree = renderer.create(<Snackbar visible={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('when {onPress}', () => {
  //   // @TODO: Test <Snackbar /> {onPress}
  // });
});
