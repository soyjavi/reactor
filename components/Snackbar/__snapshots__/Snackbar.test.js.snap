// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar> renders 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "#4a4a4a",
                },
              ],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Snackbar> when {button} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "#4a4a4a",
                },
              ],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Snackbar> when {caption} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "#4a4a4a",
                },
              ],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#4a4a4a",
                        "fontFamily": undefined,
                        "fontWeight": "400",
                        "textAlignVertical": "center",
                      },
                      Object {
                        "fontSize": 14,
                      },
                      false,
                      Object {
                        "color": "#ffffff",
                        "flex": 1,
                        "marginRight": 12,
                      },
                    ]
                  }
                >
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s.
                </Text>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Snackbar> when {children} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "#4a4a4a",
                },
              ],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s.
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Snackbar> when {color} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "#4a4a4a",
                },
              ],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Snackbar> when {visible} (false) 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "#4a4a4a",
                },
              ],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
